reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt

```diff
@@ -89,11 +89,14 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
         createExcludeInfo()
         createScopes()
 
-        input.ortResult.analyzer?.result?.projects?.forEach { project ->
+        val resultProjects = input.ortResult.getProjects().sortedBy { it.id }
```

Comment:
I'd prefer to not introduce these variables and do the sorting as part of the constructor call in line 124.
This would make it more obvious that packages and projects are actually sorted correctly: in the current code one needs to understand all the side effect the called functions like `addProject`, `addPackage` , `addDependencyTree` and `addShortestPaths` have. Also it's simpler to do the sorting separately, as all the other code would not need to worry about sorting at all.

Another option would be to not do the sorting at all in this data structure, but do it in the function which actually serializes the model.

What do you think?

Reply:
> I'd prefer to not introduce these variables and do the sorting as part of the constructor call in line 124.

Unfortunately, that's a lot more work as it's not enough to just sort (evaluated) projects and packages, but also licenses, license findings, dependency trees etc. would all need to be sorted, as they're sorted in lists, and the insertion order is determined by the iteration order over the original projects / packages.

> Another option would be to not do the sorting at all in this data structure, but do it in the function which actually serializes the model.

It would have been nice to be able to (easily) do that, but due to the complexity of the involved Jackson magic and the plethora of lists being used, it's hard to do. For example, `licenses` are not sorted by their own string representation, but by the id of the packages they stem from. IMO, the evaluated model is quite messed up in general, and I'm reluctant to fix it in the context of this change.